benefits/core/models/transit.py

```diff
@@ -27,6 +27,67 @@ def agency_logo_large(instance, filename):
     return _agency_logo(instance, filename, "lg")
 
 
+class TransitProcessorConfig(models.Model):
+    """
+    Fields common to models that represent configuration for connecting to a transit processor,
+    an entity that applies transit agency fare rules to rider transactions.
+    """
+
+    class Meta:
+        abstract = True
+
+    class Environment(models.TextChoices):
+        QA = "qa", "QA"
+        PROD = "prod", "Production"
+
+    id = models.AutoField(primary_key=True)
+    environment = models.TextField(
+        choices=Environment,
+        help_text="A label to indicate which environment this configuration is for.",
+    )
+
+
+class LittlepayConfig(TransitProcessorConfig):
+    """Configuration for connecting to Littlepay."""
+
+    agency_slug = models.SlugField(
+        choices=core_context.AgencySlug,
+        help_text="A label to indicate which agency this configuration is for. Note: the field that controls which configuration an agency actually uses is on the TransitAgency model.",  # noqa
+    )
+    audience = models.TextField(
+        help_text="This agency's audience value used to access the TransitProcessor's API.", default="", blank=True
+    )
+    client_id = models.TextField(
+        help_text="This agency's client_id value used to access the TransitProcessor's API.", default="", blank=True
+    )
+    client_secret_name = SecretNameField(
+        help_text="The name of the secret containing this agency's client_secret value used to access the TransitProcessor's API.",  # noqa: E501
+        default="",
+        blank=True,
+    )
+
+    @property
+    def client_secret(self):
+        secret_field = self._meta.get_field("client_secret_name")
+        return secret_field.secret_value(self)
+
+    def clean(self):
+        field_errors = {}
+
+        if hasattr(self, "transitagency") and self.transitagency.active:
+            message = "This field is required when this configuration is referenced by an active transit agency."
+            needed = dict(audience=self.audience, client_id=self.client_id, client_secret_name=self.client_secret_name)
+            field_errors.update({k: ValidationError(message) for k, v in needed.items() if not v})
+
+        if field_errors:
+            raise ValidationError(field_errors)
+
+    def __str__(self):
+        environment_label = self.Environment(self.environment).label if self.environment else "unknown"
+        agency_slug = self.agency_slug if self.agency_slug else "(no agency)"
+        return f"({environment_label}) {agency_slug}"
```

Comment:
Maybe to answer my own [question above](), `agency_slug` and the `__str__` implementation could be moved to the ABC?

This too could be cleaned up if part / all of it was moved into the ABC

```python
environment_label = self.Environment(self.environment).label if self.environment else "unknown"
```